apps/datahub/src/app/home/search/search-filters/search-filters.component.ts

```diff
@@ -24,14 +27,17 @@ export class SearchFiltersComponent implements OnInit {
   filters: QueryList<FilterDropdownComponent>
   searchConfig: { fieldName: string; title: string }[]
   isOpen = false
+  userId: string
 
   constructor(
     public searchFacade: SearchFacade,
     private searchService: SearchService,
-    private fieldsService: FieldsService
+    private fieldsService: FieldsService,
+    private authService: AuthService
   ) {}
 
   ngOnInit(): void {
+    this.authService.user$.subscribe((user) => (this.userId = user?.id))
```

Comment:
Storing the value of an observable in a local state is kind of an anti-pattern, and it can lead to nasty race conditions. In this case I think it's fine actually, so this can stay like that, but in an ideal world where we would have as much time as we want, this should have been a `userId$` observable 🙂 (and yes this makes everything more verbose and harder to read)